Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyDraftOrdersAPI.Codeunit.al

```diff
@@ -0,0 +1,385 @@
+        NotesTextBuilder: TextBuilder;
+    begin
+        SalesCommentLine.SetRange("Document Type", SalesCommentLine."Document Type"::"Posted Invoice");
+        SalesCommentLine.SetRange("No.", TempShpfyOrderHeader."Sales Invoice No.");
+
+        if SalesCommentLine.FindSet() then begin
+            GraphQuery.Append(', note: \"');
+            repeat
+                NotesTextBuilder.Append(SalesCommentLine.Comment + '\n');
+            until SalesCommentLine.Next() = 0;
+            GraphQuery.Append(NotesTextBuilder.ToText());
+            GraphQuery.Append('\"');
+        end;
+    end;
+
+    local procedure AddPaymentTerms(var GraphQuery: TextBuilder; var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary)
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        ShpfyPaymentTerm: Record "Shpfy Payment Terms";
+        DueAtDateTime: DateTime;
+        IssuedAtDateTime: DateTime;
+    begin
+        if not ShopifyPaymentTermsExists(ShpfyPaymentTerm, TempShpfyOrderHeader, SalesInvoiceHeader) then
+            exit;
+
+        GraphQuery.Append(', paymentTerms: {');
+        GraphQuery.Append('paymentTermsTemplateId: \"gid://shopify/PaymentTermsTemplate/');
+        GraphQuery.Append(Format(ShpfyPaymentTerm."Id"));
+        GraphQuery.Append('\"');
+
+        Evaluate(IssuedAtDateTime, Format(SalesInvoiceHeader."Document Date"));
+        Evaluate(DueAtDateTime, Format(SalesInvoiceHeader."Due Date"));
+
+        GraphQuery.Append(', paymentSchedules: {');
+        if ShpfyPaymentTerm.Type = 'FIXED' then begin
+            GraphQuery.Append('dueAt: \"');
+            GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(Format(DueAtDateTime, 0, 9)));
+            GraphQuery.Append('\"');
+        end else
+            if ShpfyPaymentTerm.Type = 'NET' then begin
+                GraphQuery.Append(', issuedAt: \"');
+                GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(Format(IssuedAtDateTime, 0, 9)));
+                GraphQuery.Append('\"');
+            end;
+
+        GraphQuery.Append('}}');
+    end;
+
+    local procedure ShopifyPaymentTermsExists(
+        var ShpfyPaymentTerms: Record "Shpfy Payment Terms";
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        var SalesInvoiceHeader: Record "Sales Invoice Header"
+    ): Boolean
+    begin
+        SalesInvoiceHeader.Get(TempShpfyOrderHeader."Sales Invoice No.");
+        ShpfyPaymentTerms.SetRange("Payment Terms Code", SalesInvoiceHeader."Payment Terms Code");
+        ShpfyPaymentTerms.SetRange("Shop Code", ShpfyShop.Code);
+
+        if not ShpfyPaymentTerms.FindFirst() then begin
+            ShpfyPaymentTerms.SetRange("Payment Terms Code");
+            ShpfyPaymentTerms.SetRange("Is Primary", true);
+
+            if not ShpfyPaymentTerms.FindFirst() then
+                exit(false);
+        end;
+
+        exit(true);
+    end;
+
+    local procedure GetISOCode(CurrencyCode: Code[10]): Code[3]
```

Comment:
I think this should be moved to where you populate CurrencyCode to the TempHeader. So here you can simply assign it. 

In the case where you take the currency code from the GenLedgSetup, you should assume that it's in ISO format. 
The chance that LCY is not in ISO format is lower than LCY being as a record in Currencies.

Reply:
Okay, I see that Andrei said you should get it from Currency. Still weird to me, but okay.